### PR TITLE
[en] Certificates

### LINKED ISSUE #527: [ English ]: Digital Certificates
Please can I contribute by adding this term to glossary ?

### PR DESCRIPTION
**Pull request by iamNoah1**
close #527

### CODE REVIEW

**Comment by iamNoah1** [comment]
Memo to myself: use semantic line breaks! @iamNoah1

File: content/en/certificate.md
Comment (CathPag) [inline content/en/certificate.md:10]:
```suggestion
A certificate — also often referred to as a digital certificate, public key certificate, or SSL certificate — is a technology used to verify the identity of an entity on the internet. An entity can be the sender of an email or a website. Additionally, certificates encrypt communication between entities on the internet.
```

File: content/en/certificate.md
Comment (CathPag) [inline content/en/certificate.md:15]:
```suggestion
Communication over the internet is unencrypted by default. Consequently, any communication can potentially be intercepted, compromising sensitive information like usernames and passwords. Think about your favorite coffee shop or a hotel lobby, where you connect to the available wifi router. Someone could get between your device and the wifi router and get hold of your credentials or read your emails and chats.
```

File: content/en/certificate.md
Comment (CathPag) [inline content/en/certificate.md:20]:
```suggestion
Certificates are a technology that, on the one side, verifies that communication partners are who they say they are — for example, an email sender or a website that you visit on your browser. Modern email clients will notify you if a sender's identity is correct, as will the browsers (notice the little lock in front of the URL). A certificate's trust derives from the certificate authority, often abbreviated as CA, that issues the certificate, guaranteeing that the certificate holder's identity is correct.
```

File: content/en/certificate.md
Comment (CathPag) [inline content/en/certificate.md:22]:
```suggestion
On the other side, certificates encrypt communication between entities on the internet. They contain a public key used to encrypt data, which can only be decrypted with a matching private key. As the name indicates, public keys are shared publicly, while private keys are kept private. Take a website, for example. Your browser encrypts data sent to the website using a public key, but only the website server knows the private key and can thus encrypt the data again. If someone intercepted the communication, it would be nearly impossible to read the data.
```

Comment (CathPag) [reply content/en/certificate.md:22]: Not sure I can follow this sentence: "Your browser encrypts data sent to the website using a public key, but only the website server knows the private key and can thus encrypt the data again."

If the browser encrypts data with a public key, anyone can encrypt it. Why does the server's private key matter? This may need a little more context...or I'm not getting it.

**Review by CathPag — COMMENTED** [review]
Made a few suggested edits and pointed out one sentence I don't really understand

**Comment by iamNoah1** [comment]
@CathPag thank you so much for the edits. My English needs to get more fluid :) 

Regarding your question. Yes exactly the public key is shared publicly, so everyone can encrypt data with it. Maybe the email example is more easy to understand. I want to sent an encrypted email to you. I use your public key to encrypt the mail. Everyone else who wants to send an encrypted email to you, could do the same. But only you can decrypt it, because only you have the matching private key :) I just recognized that I used encrypt both times ^^.

File: content/en/certificate.md
Comment (CathPag) [inline content/en/certificate.md:32]: Let me see if I understood correctly: the public key is able to encrypt but not decrypt (otherwise encrypting with a public key wouldn't make much sense as anyone could decrypt it again). To decrypt it you need the private key, is that correct?

If so, I think we need to clarify. In daily life, a key is used to close and open a door, so if it's public, it's available to anyone. If you need two keys, one to open and one to close, we need to say so. That's why I'm stumbling over this example and others might too.

**Review by CathPag — COMMENTED** [review]
One more edit and a question regarding the same section as it's not yet clear to me.

File: content/en/certificate.md
Comment (fsbaraglia) [reply content/en/certificate.md:32]: Exactly, like you explained :)

Comment (iamNoah1) [reply content/en/certificate.md:32]: Yes, you understood correctly :). I am happy for any suggestion to make it more clear :)

Comment (CathPag) [reply content/en/certificate.md:32]: How about this?

> On the other side, certificates encrypt communication between entities on the internet. Encryption and decryption happen with so-called keys. But unlike physical keys, certificates need two keys: a public one to encrypt data and a private one to decrypt it again. As the name indicates, public keys are shared publicly, while private keys are kept private. Take a website, for example. The browser encrypts data sent to the website using a public key and the website server holds the private key needed to decrypt it.

What I still don't get is: The public key is only for that website but, since it's used to encrypt data, they don't bother to make it private. Encryption through that public key can only be decrypted with that specific private key, right? If that's the case, maybe this is better?

>  On the other side, certificates encrypt communication between entities on the internet. Encryption and decryption happen with so-called keys. But unlike physical keys, certificates need two keys: a public one to encrypt data and a private one to decrypt it again. As the name indicates, public keys are shared publicly. Since they encrypt data, there is no need to hide it from the public. Private keys, on the other hand, are needed to decrypt data and are thus only held by those entities authorized to read it. Take a website, for example. The browser encrypts data sent to the website using a public key while the website server holds the private key needed to decrypt it.

**Review by CathPag — COMMENTED** [review]
Added two suggestions to clarify the two-key thing. Let me know what you think.

File: content/en/certificate.md
Comment (iamNoah1) [reply content/en/certificate.md:32]: TBH I am not sure if we really should explain asymmetric encryption in that much detail, as the focus of this article is on certificates and not on how encryption works. I mean one must not necessarily understand how asymmetric encryption works in order to understand what certificates are, right? Can we not just simply say, that certificates can be used for asymmetric encryption because they include an encryption key? 

I would love to hear another opinion. @JasonMorgan @seokho-son @jihoon-seo, maybe. 

In case we want to keep the detailed explanation here a little modification to make it more accurate again :) 

`On the other side, certificates can be used to encrypt communication between entities on the internet. Encryption and decryption happen with so-called keys. But unlike physical keys, with this kind of encryption/decryption, two keys are needed: a public one to encrypt data and a private one to decrypt it again. As the name indicates, public keys are shared publicly. Since they encrypt data, there is no need to hide it from the public. Private keys, on the other hand, are needed to decrypt data and are thus only held by those entities authorized to read it. Take a website, for example. The browser encrypts data sent to the website using a public key (which is included in the certificate of that website) while the website server holds the private key needed to decrypt it.`

Comment (CathPag) [reply content/en/certificate.md:32]: Looks good to me. At least I learned a lot from these back and forth, so thank you!

File: content/en/certificate.md
Comment (JasonMorgan) [inline content/en/certificate.md:27]: I think this detail can be omitted. It adds complexity to the definition in a way that isn't necessarily useful. 
```suggestion
```

File: content/en/certificate.md
Comment (JasonMorgan) [inline content/en/certificate.md:31]: Would we be open to removing all references to public and private keys and instead just say that certificates allow us to securely encrypt the data between two devices? Like talking about the CA it's a level of detail that isn't required to understand what a certificate does. You can see from the discussion below it's a complex topic and knowing what it does is much more important than how it does it.

**Review by JasonMorgan — DISMISSED** [review]
Looks good, made some suggestions for simplifying it.

File: content/en/certificate.md
Comment (seokho-son) [inline content/en/certificate.md:32]:
```suggestion
Your browser encrypts data sent to the website using a public key, but only the website server knows the private key and can thus decrypt the data. 
```

`again` 
maybe redundant.

File: content/en/certificate.md
Comment (seokho-son) [inline content/en/certificate.md:29]:
```suggestion
On the other side, certificates can be used to encrypt communication between entities on the internet. 
```

**Review by seokho-son — COMMENTED** [review]
@iamNoah1 Thank you for the new definition !

Please take a look my comments and apply it selectively :)

File: content/en/certificate.md
Comment (seokho-son) [inline content/en/certificate.md:24]: seems duplicated :)

Comment (iamNoah1) [reply content/en/certificate.md]: oops, this happens when you mix online editing through GitHub suggestion feature and manual editing ^^. Fixed

**Comment by seokho-son** [comment]
@iamNoah1 Just to check :) 

The title: `Certificate`
The first sentence says:  A `digital certificate` — also often referred ...

(either way I'm fine)

**Comment by iamNoah1** [comment]
> @iamNoah1 Just to check :)
> 
> The title: `Certificate` The first sentence says: A `digital certificate` — also often referred ...
> 
> (either way I'm fine)

dang ^^

File: content/en/digital-certificate.md
Comment (CathPag) [inline content/en/digital-certificate.md:10]:
```suggestion
A (digital) certificate — also often referred to as a public key certificate, or SSL certificate — is a digital document used to help secure communications over the network. 
```

Comment (CathPag) [reply content/en/digital-certificate.md:10]: I know there was a discussion about whether to call it certificate or digital certificate. How about adding parenthesis to indicate it can be used with or without digital?

**Review by CathPag — COMMENTED** [review]
Sorry, one small idea — it just occurred to me. Happy to approve asap if you have time to accept or decline real quick

**Comment by iamNoah1** [comment]
> Sorry, one small idea — it just occurred to me. Happy to approve asap if you have time to accept or decline real quick

done @CathPag :)

**Review by CathPag — APPROVED** [review]
LGTM

**Pull request merged**
